File: 268_MissingNumber.py
class Solution(object):
    def missingNumber(self, nums):
        """
        Given an array containing n distinct numbers taken from 0, 1, 2, ..., n, find the one that is missing from the array.
        For example,
        Given nums = [0, 1, 3] return 2.
        Note:
        Your algorithm should run in linear runtime complexity. Could you implement it using only constant extra space complexity?
        :type nums: List[int]
        :rtype: int
        """
        l = len(nums)
        return (l + 1) * l / 2 - sum(nums)

File: 268.MissingNumber.java
import java.util.Arrays;

class Solution {
    public int missingNumber(int[] nums) {
        int result = 0;
        Arrays.sort(nums);
        for (int i = 0; i < nums.length + 1; i += 1) {
            if (i == nums.length) {
                return nums.length;
            }
            if (i != nums[i]) {
                result = i;
                break;
            }
        }
        return result;
    }
}